index.js

```diff
@@ -9,18 +12,29 @@ const EDITOR_MODULE = {
   worker: {
     id: 'vs/editor/editor',
     entry: 'vs/editor/editor.worker',
-    output: 'editor.worker.js',
     fallback: undefined
   },
   alias: undefined,
 };
 const LANGUAGES = require('./languages');
 const FEATURES = require('./features');
 
+/**
+ * Return a resolved path for a given Monaco file.
+ */
 function resolveMonacoPath(filePath) {
   return require.resolve(path.join('monaco-editor/esm', filePath));
 }
 
+/**
+ * Return the interpolated final filename for a worker, respecting the file name template.
+ */
+function getWorkerFilename(filename, entry) {
+  return loaderUtils.interpolateName({resourcePath: entry}, filename, {
+    content: fs.readFileSync(resolveMonacoPath(entry))
```

Comment:
Using `loader-utils` so that the full breadth of Webpack `filename` options is supported. We read the entry file from the file system so that content-based hashes will update based on worker file contents.